lxd/storage/backend_lxd.go

```diff
@@ -1172,6 +1172,8 @@ func (b *lxdBackend) RefreshCustomVolume(projectName string, srcProjectName stri
 
 	if contentDBType == db.StoragePoolVolumeContentTypeBlock {
```

Comment:
This is starting to feel like it would be clearer as a switch statement with a default case of returning an error with an unsupported content type error.

Reply:
Here we could it. You were thinking of something like this, right?

```go
	contentDBType, err := VolumeContentTypeNameToContentType(srcConfig.Volume.ContentType)
	if err != nil {
		return err
	}

	// Get the source volume's content type.
	var contentType drivers.ContentType

	switch contentDBType {
	case db.StoragePoolVolumeContentTypeFS:
		contentType = drivers.ContentTypeFS
	case db.StoragePoolVolumeContentTypeBlock:
		contentType = drivers.ContentTypeBlock
	default:
		return fmt.Errorf("Unsupported content type")
	}
```

Reply:
Yeah, thinking about my above comment more, you could do this:

```go
contentType, err := VolumeDBContentTypeToContentType(contentDBType)
if err != nil {
    return err
}

if contentType != db.StoragePoolVolumeContentTypeFS && contentType != db.StoragePoolVolumeContentTypeBlock {
    		return fmt.Errorf("Volume of content type %q cannot be refreshed", contentType)
}
```